spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/eureka/EurekaDiscoveryClientConfiguration.java

```diff
@@ -174,4 +175,18 @@ public EurekaHealthIndicator eurekaHealthIndicator(EurekaClient eurekaClient,
 		}
 	}
 
+	@Configuration
+	@ConditionalOnProperty(value = "eureka.client.healthcheck", matchIfMissing = false)
```

Comment:
Can you make it `eureka.client.healthcheck.enabled`? That matches our convention a bit better.